psi4/src/psi4/libmints/molecule.cc

```diff
@@ -359,9 +359,11 @@ void Molecule::add_unsettled_atom(double Z, std::vector<std::string> anchor, std
 
 
 
-double Molecule::mass(int atom) const {
+double Molecule::mass(int atom, bool zero_ghost) const {
     double ret = 0.0;
-    if (atoms_[atom]->mass() != 0.0)
+    if (zero_ghost && (atoms_[atom]->Z() == 0.0))
```

Comment:
I'm not totally happy with checking `CoordEntry::Z` to check for Type A (Gh(x)) ghosting, but `is_ghosted` only works for type B (`extract_subsets`) ghosting. So, this is the only way that gives consistent behavior for both types to my knowledge.

Reply:
If Checking `CoordEntry::Z` is the only way to catch both type A and type B ghosts then its fine. Changing the behavior of `CoordEntry` is far more complicated and outside the scope of this PR